pagination/helpers.go

```diff
@@ -0,0 +1,54 @@
+package pagination
+
+import (
+	"encoding"
+	"encoding/base64"
+)
+
+// PageSizeGetter holds a method to return the amount of pages requested by users when listing items in an API call.
+type PageSizeGetter interface {
+	// GetPageSize returns the desired page size.
+	GetPageSize() int32
+}
+
+// PageSize returns a valid page size following the AIP-158 for Pagination.
+// Reference: https://google.aip.dev/158
+//
+//	Default value: 50
+//	Max value: 1000
+//	Min value: 0.
+//
+//	If no value is passed, it returns 50.
+//	If a value greater than 1000 is specified, it caps the result value to 1000.
+//	If a negative value is specified, it returns -1.
+func PageSize(req PageSizeGetter) int32 {
+	if req == nil {
+		return defaultPageSize
+	}
+	if req.GetPageSize() == 0 {
+		return defaultPageSize
+	}
+
+	if req.GetPageSize() > maxPageSize {
+		return maxPageSize
+	}
+	if req.GetPageSize() < 0 {
+		return -1
+	}
+	return req.GetPageSize()
+}
```

Comment:
This is a great proposal, although I think we should defer it to a next iteration.

Reply:
I don't think it will happen if we don't do it now.

We can use variable arguments and only use the first element of the array to literally not have to modify any of the existing calls in other PRs.

Reply:
Thanks for the comment, I've added a change covering page size options!